tensor2tensor/rl/player_utils.py

```diff
@@ -0,0 +1,330 @@
+        low=0,
+        high=255,
+        shape=extended_shape,
+        dtype=np.uint8)
+
+  def observation(self, frame):
+    """Add single zero row/column to observation if needed."""
+    if frame.shape == self.observation_space.shape:
+      return frame
+    else:
+      extended_frame = np.zeros(self.observation_space.shape,
+                                self.observation_space.dtype)
+      assert self.HW_AXES == (0, 1)
+      extended_frame[:frame.shape[0], :frame.shape[1]] = frame
+      return extended_frame
+
+  def if_odd(self, n):
+    return n % 2
+
+
+class RenderObservations(gym.Wrapper):
+  """Add observations rendering in 'rgb_array' mode."""
+  def __init__(self, env):
+    super(RenderObservations, self).__init__(env)
+    if "rgb_array" not in self.metadata["render.modes"]:
+      self.metadata["render.modes"].append("rgb_array")
+
+  def step(self, action):
+    ret = self.env.step(action)
+    self.last_observation = ret[0]
+    return ret
+
+  def reset(self, **kwargs):
+    self.last_observation = self.env.reset(**kwargs)
+    return self.last_observation
+
+  def render(self, mode="human", **kwargs):
+    assert mode == "rgb_array"
+    return self.last_observation
+
+
+def wrap_with_monitor(env, video_dir):
+  """Wrap environment with gym.Monitor
+
+  Video recording provided by Monitor requires
+    1) both height and width of observation to be even numbers.
+    2) rendering of environment
+  """
+  env = ExtendToEvenDimentions(env)
+  env = RenderObservations(env)  # pylint: disable=redefined-variable-type
+  env = wrappers.Monitor(env, video_dir, force=True,
+                         video_callable=lambda idx: True,
+                         write_upon_reset=True)
+  return env
+
+
+def create_simulated_env(
+    output_dir, grayscale, resize_width_factor, resize_height_factor,
+    frame_stack_size, generative_model, generative_model_params,
+    random_starts=True, which_epoch_data="last", **other_hparams
+):
+  """"Create SimulatedEnv with minimal subset of hparams."""
+  # We need these, to initialize T2TGymEnv, but these values (hopefully) have
+  # no effect on player.
+  a_bit_risky_defaults = {
+      "game": "pong",  # assumes that T2TGymEnv has always reward_range (-1,1)
+      "real_batch_size": 1,
+      "rl_env_max_episode_steps": -1,
+      "max_num_noops": 0
+  }
+
+  for key in a_bit_risky_defaults:
+    if key not in other_hparams:
+      other_hparams[key] = a_bit_risky_defaults[key]
+
+  hparams = tf.contrib.training.HParams(
+      grayscale=grayscale,
+      resize_width_factor=resize_width_factor,
+      resize_height_factor=resize_height_factor,
+      frame_stack_size=frame_stack_size,
+      generative_model=generative_model,
+      generative_model_params=generative_model_params,
+      **other_hparams
+  )
+  return SimulatedEnv(output_dir, hparams, which_epoch_data=which_epoch_data,
+                      random_starts=random_starts)
+
+
+class PPOPolicyInferencer(object):
```

Comment:
There are two different interfaces: 
- `PolicyLearner` is heavy - requires simulated env (not in `_init_` thought), performs high level training and evaluation.
- `PPOPolicyInferencer` only loads policy from disk, provides low level, non-tf interface for policy inference.

`PolicyLearner` also infers policy, but does it inside of graph, whole idea of `PPOPolicyInferencer` is to provide python api for different purposes.

All the methods are for these reasons different. We could merge these classes, this would require removing `base_event_dir`, `total_num_epochs` arguments from `PolicyLearner.__init__` (or make them optional), initialize inference graph in `PPOLearner.__init__` (but this will be unused in rlmb pipeline). I'm not sure if that will make things more clear. We can also just move `PPOPolicyInferencer` to `ppo_learner.py`. Do you think it would suffice?